Misc/NEWS.d/next/Library/2024-12-02-21-19-53.gh-issue-127544.N7uEsX.rst

```diff
@@ -0,0 +1,2 @@
+Ensure that ``AF_DECnet`` is part of the :class:`!socket.AddressFamily`
```

Comment:
The reason why `socket.AddressFamily` does not have a link is because... we don't document it.

Reply:
ah, okay. Should I not have a NEWS entry for this then, since it's modifying an undocumented class by adding an even more undocumented member?

Reply:
We don't document it *but* we refere to it... so I'm not really sure. I think it's fine that way (see https://docs.python.org/3/library/socket.html#constants).

What you can however do is to add it to the global `socket.*`

Reply:
I'm not sure what you mean by "add it to the global `socket.*`. If you're talking about `socket.AF_DECnet`, that's already covered. It's imported as an int value right now, and `IntEnum._convert_` is a magic method that will replace the existing object with the member of `AddressFamily` once it's created.

Reply:
> If you're talking about socket.AF_DECnet, that's already covered. It's imported as an int value right now

I meant as a documented entry.

Reply:
ah okay, I understand

Reply:
Actually there's quite a lot of `AF_*` constants which aren't documented:

```
AF_APPLETALK
AF_ASH
AF_ATMPVC
AF_ATMSVC
AF_AX25
AF_BRIDGE
AF_DECnet
AF_ECONET
AF_IPX
AF_IRDA
AF_KEY
AF_LLC
AF_NETBEUI
AF_NETROM
AF_PPPOX
AF_ROSE
AF_ROUTE
AF_SECURITY
AF_SNA
AF_SYSTEM
AF_WANPIPE
AF_X25
```

I don't think AF_DECnet needs to move off that list.

Reply:
Oh, ok. Then forget my comment!